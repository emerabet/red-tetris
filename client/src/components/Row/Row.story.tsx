import React from 'react';
import { storiesOf } from '@storybook/react';
import Row from './Row';

storiesOf('Row', module)
    .add('empty', () => (
        <Row
            row={[0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
            index={3}
        />
    )).add('333', () => (
        <Row
            row={[0, 0, 0, 3, 3, 3, 0, 0, 0, 0]}
            index={3}
        />
    ));
