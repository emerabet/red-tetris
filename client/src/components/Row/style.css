.boardRow {
	display: flex;
	flex-direction: row;
}
