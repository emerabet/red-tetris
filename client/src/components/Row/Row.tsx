import React, { Component, useState, useEffect } from 'react';
import Cell from '../Cell';

import './style.css';

interface RowProps {
  row: number[];
  index: number;
}

const Row: React.SFC<RowProps> = (props) => {
  return (
    <div key={props.index} className="boardRow">
      {
        props.row.map((c: number, j: number) => {
          return (
            <Cell key={`cell_${j}`} index={j} color={c} />
          );
        })
      }
    </div>
  );
};

export default Row;
