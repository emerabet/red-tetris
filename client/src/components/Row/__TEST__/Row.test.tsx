import React from 'react';

import renderer from 'react-test-renderer';
import Row from '../Row';

it('renders correctly with defaults', () => {
  const row = renderer.create(<Row
    row={[0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}
            index={3}
  />).toJSON();
  expect(row).toMatchSnapshot();
});
