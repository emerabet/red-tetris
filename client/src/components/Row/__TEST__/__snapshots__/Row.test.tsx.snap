// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<div
  className="boardRow"
>
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
  <div
    className="boardCell cell0"
  />
</div>
`;
